packages/live-preview-sdk/src/inspectorMode/utils.ts

```diff
@@ -271,3 +281,53 @@ export function getAllTaggedEntries({
     ),
   ];
 }
+
+const isElementOverlapped = (element: Element, coordinates: DOMRect, root = window.document) => {
+  const { top, right, bottom, left } = coordinates;
+  const topLeft = root.elementFromPoint(left, top);
+  const topRight = root.elementFromPoint(right, top);
+  const bottomLeft = root.elementFromPoint(left, bottom);
+  const bottomRight = root.elementFromPoint(right, bottom);
+
+  return (
+    topLeft === element || topRight === element || bottomLeft === element || bottomRight === element
+  );
+};
+
+const addVisibilityToTaggedElements = (
+  taggedElements: Partial<TaggedElement>[],
+  root = window.document,
+) =>
+  taggedElements.map((taggedElement) => ({
+    ...taggedElement,
+    isVisible: taggedElement.element!.checkVisibility({
+      checkOpacity: true,
+      checkVisibilityCSS: true,
+    }),
+    isCoveredByOtherElement: !isElementOverlapped(
+      taggedElement.element!,
+      taggedElement.coordinates!,
+      root,
+    ),
+  }));
+
+const addCoordinatesToTaggedElements = (
+  taggedElements: Partial<TaggedElement>[],
+): Partial<TaggedElement>[] =>
+  taggedElements.map(({ element, attributes }) => ({
+    element,
+    coordinates: element!.getBoundingClientRect(),
+    attributes,
+  }));
+
+/**
+ * Applies the attributes that we cannot simply get from the tagged elements itself
+ * but need to calculate based on the current state of the document
+ */
+export const addCalculatedAttributesToTaggedElements = (
+  taggedElements: PrecaulculatedTaggedElement[],
+  root = window.document,
+): TaggedElement[] => {
```

Comment:
```suggestion
): PrecaulculatedTaggedElement[] => {
```

Reply:
same then in 332

Reply:
No this should use the regular one as it now has added the calculations

Reply:
I've merged for now, let me know I can follow up later on this if you disagree